verify/common.go

```diff
@@ -35,146 +35,351 @@ type ErrWithHelp interface {
+		context.TODO(),
+		owner,
+		repo,
+		checkRunID,
+		github.UpdateCheckRunOptions{
+			Name:    p.Name,
+			HeadSHA: github.String(headSHA),
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("unable to update check result: %w", err)
+	}
 
-	resRun, updateResp, err := env.Client.Checks.UpdateCheckRun(context.TODO(), orgName, repoName, resRun.GetID(), github.UpdateCheckRunOptions{
-		Name: p.Name,
-		Status: github.String("completed"),
-		Conclusion: github.String(conclusion), 
-		CompletedAt: &completedAt,
+	p.debugf("update check API response: %+v", updateResp)
+	p.debugf("updated run: %+v", checkRun)
+
+	return nil
+}
+
+func (p PRPlugin) finishCheckRun(client *github.Client, owner, repo string, checkRunID int64, conclusion, summary, text string) error {
+	p.debugf("updating check run %q on %s/%s...", p.Name, owner, repo)
+
+	checkRun, updateResp, err := client.Checks.UpdateCheckRun(context.TODO(), owner, repo, checkRunID, github.UpdateCheckRunOptions{
+		Name:        p.Name,
+		Conclusion:  github.String(conclusion),
+		CompletedAt: &github.Timestamp{Time: time.Now()},
 		Output: &github.CheckRunOutput{
-			Title: github.String(p.Title),
+			Title:   github.String(p.Title),
 			Summary: github.String(summary),
-			Text: github.String(fullHelp),
+			Text:    github.String(text),
 		},
 	})
 	if err != nil {
 		return fmt.Errorf("unable to update check result: %w", err)
 	}
 
-	env.Debugf("update check API response: %+v", updateResp)
-	env.Debugf("updated run: %+v", resRun)
+	p.debugf("update check API response: %+v", updateResp)
+	p.debugf("updated run: %+v", checkRun)
+
+	return nil
+}
+
+func (p *PRPlugin) Entrypoint(env *ActionsEnv) (err error) {
+	switch env.Event.GetAction() {
+	case "open":
+		err = p.onOpen(env)
+	case "reopen":
+		err = p.onReopen(env)
+	case "edit":
+		err = p.onEdit(env)
+	case "synchronize":
+		err = p.onSync(env)
+	default:
+		// Do nothing
+	}
+
+	return
+}
+
+func (p PRPlugin) onOpen(env *ActionsEnv) error {
+	headSHA := env.Event.GetPullRequest().GetHead().GetSHA()
+
+	// Create the check run
+	checkRun, err := p.createCheckRun(env.Client, env.Owner, env.Repo, headSHA)
+	if err != nil {
+		return err
+	}
 
-	// return failure here too so that the whole suite fails (since the actions
+	// Process the PR
+	conclusion, summary, text, procErr := p.processPR(env.Event.PullRequest)
+
+	// Update the check run
+	if err := p.finishCheckRun(env.Client, env.Owner, env.Repo, checkRun.GetID(), conclusion, summary, text); err != nil {
+		return err
+	}
+
+	// Return failure here too so that the whole suite fails (since the actions
 	// suite seems to ignore failing check runs when calculating general failure)
 	if procErr != nil {
 		return fmt.Errorf("failed: %v", procErr)
 	}
 	return nil
 }
 
-type ActionsEnv struct {
-	Event *github.PullRequestEvent
-	Client *github.Client
+func (p PRPlugin) onReopen(env *ActionsEnv) error {
+	headSHA := env.Event.GetPullRequest().GetHead().GetSHA()
```

Comment:
re-open should re-run the check, IMO, since IIRC the title can change (plus, it's useful in case things go wrong somehow)

Reply:
I've seen how `edit` events are spread even if the PR is closed, so the `edit` event should update the checks and `reopen` just fetch the global result. I wanted to double-check this but I don't know if I can from a branch different than master. Because master is used for this repo (`kubernetes-sigs/kubebuilder-release-tools`), right?